cmake/stm32/h5.cmake

```diff
@@ -0,0 +1,23 @@
+set(STM32_H5_TYPES 
+    H503xx H562xx H563xx H573xx
+)
+
+set(STM32_H5_TYPE_MATCH 
+    "H503.." "H562.." "H563.." "H573.."
+)
+set(STM32_H5_RAM_SIZES 
+    32K 640K 640K 640K
+)
+set(STM32_H5_CCRAM_SIZES 
+    0K  0K  0K  0K
+)
+
+stm32_util_create_family_targets(H5)
+
+target_compile_options(STM32::H5 INTERFACE 
+    -mcpu=cortex-m33 -mfloat-abi=hard -mfpu=fpv5-d16 -mthumb
+)
+
+target_link_options(STM32::H5 INTERFACE 
+    -mcpu=cortex-m33 -mfloat-abi=hard -mfpu=fpv5-d16 -mthumb
```

Comment:
What is the source information for using fpv5-d16 FPU?

I was kinda not sure how to find this out so I opted to let STM32CubeMx generate an empty makefile project for STM32H5 and checked it's compiler options. CubeMx used fpv4-sp-d16 there.

Reply:
I could not find the answer easily. For what I can tell https://www.st.com/resource/en/programming_manual/pm0264-stm32-cortexm33-mcus-programming-manual-stmicroelectronics.pdf states that m33 devices support fpv5 instructions...
Not quite sure for this one... @atsju any input ?

Reply:
Sorry for delay. I don't have much time right now so I just answer the question here.

I have read this and others: https://community.st.com/t5/stm32cubemx-mcus/bug-in-cubemx-generating-makefile-missing-some-cpu-info/m-p/110239

For me correct answer is `fpv5-sp-d16`. v5 as pointed out by @hish15 but sp only as cubeMx and datasheets stating it's single point FPU. Do not hesitate to redo a test with latest cubeMx and to ask ST